### ISSUE TITLE
on_publish 3xx has a problem

### ISSUE THREAD

**Issue by alinasrin** [arut/nginx-rtmp-module#1307]
Hello,
I use on_publish to get new rtmp link which push stream to new link.
on page i send header("HTTP/1.1 302 publish");        header("Location:rtmp://xxx.xxx.xxx.xxx/rtmp/32873453765?s_efg=eyJxZV9ncm91cHMiOnsiaWdfbGl2ZV9lbmFibGVfbGl2ZWRhc2giOnsibGl2ZWRhc2hfbW9kZSI6InByaW1hcnkifSwiaWdfZmJsaXZlX3NlcnZlcl9leHBlcmltZW50Ijp7InNlcnZpY2VfcG9ydCI6IjE3MDAwIn0sImlnX2xvd19sYXRlbmN5X3Byb2R1Y3Rpb25fdW5pdmVyc2UiOnsiZW5hYmxlZCI6IjEifSwiaWdfbGl2ZV9lbmFibGVfcG9wX3ByaW1pbmciOnsiZW5hYmxlZCI6IjAifX19&s_vt=ig&a=ATgrKuuGczE3Q1g0");

but new stream does not work.

i check error log , in error log location is wrong:
rtmp://xxx.xxx.xxx.xxx/rtmp/32873453765?s_efg=eyJxZV9ncm91cHMiOnsiaWdfbGl2ZV9lbmFibGVfbGl2ZWRhc2giOnsibGl2ZWRhc2hfbW9kZSI6InByaW1hcnkifSwiaWdfZmJsaXZlX3NlcnZlcl9leHBlcmltZW50Ijp7InNlcnZpY2VfcG9ydCI6IjE3MDAwIn0sImlnX2xvd19sYXRlbmN5X3Byb2R1Y3Rpb25fdW5pd

The rest of the characters have been deleted.
Anyone have a solution?

**Comment by winshining** [comment]
The length of `args` variable in project is 256, your argument is too long to fit it, so it was truncated.

**Comment by alinasrin** [comment]
@winshining 
Thank you for answer, can i increase the length of args?
Unfortunately i cant decrease the length of argument.

**Comment by winshining** [comment]
Yes. In file ngx_rtmp_cmd_module.h, you can change the value of macro NGX_RTMP_MAX_NAME.
Don't forget to rebuild the module.
